src/main/views/webContentEvents.ts

```diff
@@ -206,17 +207,17 @@ export const addWebContentsEventListeners = (mmview, getServersFunction) => {
     }
 
     const willNavigate = generateWillNavigate(getServersFunction);
-    contents.on('will-navigate', willNavigate);
+    contents.on('will-navigate', willNavigate as (e: Event, u: string) => void); // TODO: Electron types don't include sender for some reason
```

Comment:
maybe the type is [IpcRendererEvent](https://www.electronjs.org/docs/latest/api/structures/ipc-renderer-event)? it should include the sender since their api is actually mentioning it. If they don't: it would be a doc bug.

Reply:
It's a bug in their types. They expect an `Electron.Event` type as the first parameter of the `will-navigate` which doesn't include `sender`, even though it's being passed in.

From `electron.d.ts`:
```
on(event: 'will-navigate', listener: (event: Event,
                                          url: string) => void): this;
    once(event: 'will-navigate', listener: (event: Event,
                                          url: string) => void): this;
    addListener(event: 'will-navigate', listener: (event: Event,
                                          url: string) => void): this;
    removeListener(event: 'will-navigate', listener: (event: Event,
                                          url: string) => void): this;
```